core/src/main/java/io/ddf/analytics/AggregationHandler.java

```diff
@@ -122,24 +122,31 @@ public DDF groupBy(List<String> groupedColumns) {
     return this.getDDF();
   }
 
+  protected String buildGroupBySQL(List<String> aggregateFunctions) {
+    String groupedColSql = Joiner.on(",").join(mGroupedColumns);
+
+    String selectFuncSql = convertAggregateFunctionsToSql(aggregateFunctions.get(0));
+    for (int i = 1; i < aggregateFunctions.size(); i++) {
+      selectFuncSql += "," + convertAggregateFunctionsToSql(aggregateFunctions.get(i));
```

Comment:
Not completely, for now we can use Guava's List.transform (i.e. Scala map), http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Lists.html#transform(java.util.List,%20com.google.common.base.Function)